### ISSUE TITLE
使用ffmpeg-rockchip对H264数据进行硬件解码失败

### ISSUE THREAD

**Issue by CETC-AIoT** [nyanmisaka/ffmpeg-rockchip#113]
硬件平台：RK3568
内核：4.19.192/5.10.198
mpp版本：2023-12-14版本及最新版本

使用avcodec_send_packet方法（h264_rkmpp）进行解码，其中avpacket是通过av_parser_parse2方法解析内存得到的，
但是这个方法返回报错，先是返回-11，然后返回 -542398533（Generic error in an external library.）
把h264码流保存为文件，然后使用mpi_dec_test进行解码，则有些帧可以解，大部分解不了，显示err 10 discard 0.
而使用AV_CODEC_ID_H264对avpacket进行软解则没有问题，请问一下是什么问题？是mpp的兼容性问题么？
另外使用软件的时候，有时解码之后的格式是NV12，有时候是YUV420P，请问这个是怎么设置的？

h264视频文件：
[video.zip](https://github.com/user-attachments/files/17096667/video.zip)

**Comment by nyanmisaka** [comment]
The attachment link you provided is invalid.

**Comment by CETC-AIoT** [comment]
您好！已经重新上传，麻烦帮忙看一下是什么问题，十分感谢！

**Comment by nyanmisaka** [comment]
MPP can't tolerate this bitstream. There are too many errors in it. So it's not ffmpeg's fault.

`mpp[150250]: h264d_sei: seq_parameter_set_id 0 may be invalid`

```
[h264 @ 0xaaaad2d01310] Format h264 detected only with low score of 1, misdetection possible!
[h264 @ 0xaaaad2d020f0] non-existing PPS 0 referenced
    Last message repeated 1 times
[h264 @ 0xaaaad2d020f0] decode_slice_header error
[h264 @ 0xaaaad2d020f0] no frame!
[h264 @ 0xaaaad2d020f0] SEI type 1 size 180 truncated at 146
[h264 @ 0xaaaad2d020f0] non-existing PPS 0 referenced
[h264 @ 0xaaaad2d020f0] SEI type 1 size 180 truncated at 144
[h264 @ 0xaaaad2d020f0] non-existing PPS 0 referenced
[h264 @ 0xaaaad2d020f0] decode_slice_header error
[h264 @ 0xaaaad2d020f0] no frame!
...
```

**Comment by nyanmisaka** [comment]
On the other hand, the fact that `mpi_dec_test` cannot decode most frames already points to the problem.

Unlike software decoders, hardware decoders are usually more strict about the bitstream.

**Comment by CETC-AIoT** [comment]
Thanks for your reply. 
I found a strange thing, I skipped the first 60 frames of data, and then I started decoding, and there was a high probability of errors, and after about 140 EAGAIN errors, I would report "Generic error in an extenal library", but I started decoding from the first frame, The probability of success is much higher, and once the data behind the success can be correctly decoded. 
In addition, I saved the data twice to a file, and then decoded it using mpi_dec_test. The result was also random, and not every frame was decoded correctly. 
GOP=30 in video stream, AUD information at the end of each frame - fixed 6 bytes: 0x00 0x00 0x00 0x01 0x09 0x10
The file is too big. I sent it to your email.
the error log:
![2a7aabe611c84f29b969650d16da8580](https://github.com/user-attachments/assets/7e8c9e3e-7027-4f36-81e0-dd7dacbcc91a)
